File: android/src/main/java/com/bishare/protocol/util/CodeGenerator.kt
package com.bishare.protocol.util

import com.bishare.protocol.constants.BIShareConfig

/**
 * Generates random codes for rooms.
 */
object CodeGenerator {

    /** Generate a random room code (4 characters). */
    fun generateRoomCode(): String = generateCode(BIShareConfig.ROOM_CODE_LENGTH)

    /** Format a room code for display (e.g., "AB3X"). */
    fun formatRoomCode(code: String): String = code.uppercase()

    private fun generateCode(length: Int): String {
        val chars = BIShareConfig.CODE_CHARSET
        return (1..length)
            .map { chars.random() }
            .joinToString("")
    }
}

File: ios/Sources/BIShareProtocol/Utilities/CodeGenerator.swift
import Foundation

/// Generates random codes for rooms.
public enum CodeGenerator {

    /// Generate a random room code (4 characters).
    public static func generateRoomCode() -> String {
        generateCode(length: BIShareConfig.roomCodeLength)
    }

    /// Format a room code for display (e.g., "AB3X").
    public static func formatRoomCode(_ code: String) -> String {
        code.uppercased()
    }

    private static func generateCode(length: Int) -> String {
        let chars = Array(BIShareConfig.codeCharset)
        return String((0..<length).map { _ in chars.randomElement()! })
    }
}
